packages/react-server/src/ReactFlightHooks.js

```diff
@@ -138,18 +118,22 @@ function use<T>(usable: Usable<T>): T {
         thenableState = createThenableState();
       }
       return trackUsedThenable(thenableState, thenable, index);
-    } else if (usable.$$typeof === REACT_SERVER_CONTEXT_TYPE) {
-      const context: ReactServerContext<T> = (usable: any);
-      return readContext(context);
+    } else if (usable.$$typeof === REACT_CONTEXT_TYPE) {
+      unsupportedContext();
     }
   }
 
-  if (__DEV__) {
-    if (isClientReference(usable)) {
-      console.error('Cannot use() an already resolved Client Reference.');
+  if (isClientReference(usable)) {
```

Comment:
I opted to keep these in PROD because this codepath always throws anyway, but the thrown error message was too obscure in this case, and it's unfortunate if the error message is different between DEV and PROD.